### DISCUSSION TITLE
Jupyter Notebook Not Running Code

### DISCUSSION THREAD

**Discussion by kingjordan3** [cooperrc/computational-mechanics#223] [Q&A]
I am having some trouble with Jupyter notebooks after a while of working on a specific notebook my code will stop running and the tab icon will change from a notebook to an hourglass and I have to keep making new notebooks to be able to run my code. Does anyone know how to fix this?

**Comment by jackseitz** [comment]
Could be a memory/CPU issue... Jupyter notebooks can consume a lot of resources/memory to run more complex calculations. Could also help to restart the kernel, sometimes it can go idle and be unresponsive. Be careful though bc this might cause you to lose variables/memory in the code... not entirely sure though. Hope this maybe leads you in the right direction, sounds like an annoying problem

**Reply by kingjordan3** [reply]
Thank you, now that I think about it I had a lot of tabs open hopefully that was the problem.

**Reply by cooperrc** [reply]
Sometimes it has to do with your computer going to sleep/losing internet connection. 

It usually helps to click "Kernel" -> "restart" or "restart and run all"
